File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        arr = []
        string = ""

        for i in range(len(s)):
            if s[i] != " ":
                string = string + s[i]
            else:
                if string != "":
                    arr.append(string)
                string = ""

        arr.append(string) if string != "" else None

        result = " ".join(arr[::-1])

        return result

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> words;
        string temp;

        for (char el : s) {
            if (el != ' ') {
                temp += el;
            } else if (!temp.empty()) {
                words.push_back(temp);
                temp.clear();
            }
        }
        
        if (!temp.empty()) {
            words.push_back(temp);
        }

        reverse(words.begin(), words.end());

        string result;
        for (int i = 0; i < words.size(); ++i) {
            result += words[i];
            if (i < words.size() - 1) result += ' ';
        }

        return result;
    }
};
